packages/logger/src/logger.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This is the file that people using Node.js will actually import. You should
+ * only include this file if you have something specific about your
+ * implementation that mandates having a separate entrypoint. Otherwise you can
+ * just use index.ts
+ */
+
+/**
+ * A container for all of the Logger instances
+ */
+export const instances: Logger[] = [];
+
+/**
+ * The JS SDK supports 5 log levels and also allows a user the ability to
+ * silence the logs altogether.
+ *
+ * The order is a follows:
+ * DEBUG < VERBOSE < INFO < WARN < ERROR < SILENT
```

Comment:
I think VERBOSE is supposed to be more verbose than DEBUG?  Honestly, I would just drop VERBOSE unless we find ourselves needing it for some reason.  If the meaning wasn't intuitive for you, it won't be intuitive for users either.

Reply:
@salqadri has a doc internally on the different levels and their use cases. There is a clear delineation between these

Reply:
AFAICT, his doc (and the iOS implementation [here](https://github.com/firebase/firebase-ios-sdk/blob/14ea068f5fd03a658017f8472a4078a727fabc3a/Firebase/Core/Public/FIRLoggerLevel.h#L20) has: DEBUG < INFO < NOTICE < WARNING < ERROR.  So to match that we would get rid of VERBOSE and add NOTICE.  Personally I don't think NOTICE adds any value though and just makes it harder to decide what level to use (both for us and our users), so I'd still be inclined to just drop VERBOSE and have: DEBUG < INFO < WARN < ERROR < SILENT.

Reply:
I've renamed the `VERBOSE` level to `LOG` (to match the console function) and have updated the doc as well. I think there is still clear use cases that we miss if we don't have `LOG` (e.g. The use case where it is **not** debug information, but the user should know, and there isn't a concrete action they can take). I'm going to leave as is.